client/state/sites/selectors/get-woo-express-trial-expiration.ts

```diff
@@ -0,0 +1,21 @@
+import { PLAN_ECOMMERCE_TRIAL_MONTHLY } from '@automattic/calypso-products';
+import { getPurchases } from 'calypso/state/purchases/selectors';
+import type { AppState } from 'calypso/types';
+
+/**
+ * Returns the expiration date of the Woo Express trial. If the trial is not active, returns null.
+ *
+ * @param {AppState} state - Global state tree
+ * @returns {string|null}
+ */
+export default function getWooExpressTrialExpiration( state: AppState ) {
```

Comment:
Do we want to explicitly document the return type in the function definition rather than the JSDoc?
```suggestion
export default function getWooExpressTrialExpiration( state: AppState ): string | null {
```